submissions/Souka21/solution.py

```diff
@@ -0,0 +1,82 @@
+import csv
+
+MAX_PRICE = 100.00
+NUM_CHEAPEST_PRODUCTS = 5
+
+def process_city_data(filename):
+    """Process input CSV file to extract city data.
+
+    Args:
+        filename (str): The name of the input CSV file.
+
+    Returns:
+        dict: A dictionary containing city data indexed by city names.
+            Each city entry contains:
+                'total_price': Total price of all products in the city.
+                'cheapest_products': List of tuples containing the names and prices of the cheapest products.
+    """
+    cities = {}
+
+    with open(filename, 'r') as file:
```

Comment:
This will load the whole file in memory and would probably trigger an OOM

Reply:
aaah ok you are right

Reply:
i can use csv.reader to prevent this